gateway-server/src/main/java/org/apache/knox/gateway/topology/monitor/DefaultConfigurationMonitorProvider.java

```diff
@@ -16,16 +16,60 @@
  */
 package org.apache.knox.gateway.topology.monitor;
 
+import java.io.File;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
 import org.apache.knox.gateway.config.GatewayConfig;
-import org.apache.knox.gateway.services.config.client.RemoteConfigurationRegistryClientService;
+import org.apache.knox.gateway.services.GatewayServices;
+import org.apache.knox.gateway.services.ServiceLifecycleException;
+import org.apache.knox.gateway.services.ServiceType;
+import org.apache.knox.gateway.services.factory.AbstractServiceFactory;
+import org.apache.knox.gateway.services.security.AliasService;
+import org.apache.knox.gateway.services.security.AliasServiceException;
+import org.apache.knox.gateway.topology.monitor.db.DbRemoteConfigurationMonitor;
+import org.apache.knox.gateway.topology.monitor.db.LocalDirectory;
+import org.apache.knox.gateway.topology.monitor.db.RemoteConfigDatabase;
+import org.apache.knox.gateway.util.JDBCUtils;
 
 
-public class DefaultConfigurationMonitorProvider implements RemoteConfigurationMonitorProvider {
+public class DefaultConfigurationMonitorProvider extends AbstractServiceFactory {
+    @Override
+    protected RemoteConfigurationMonitor createService(GatewayServices gatewayServices,
+                                    ServiceType serviceType,
+                                    GatewayConfig gatewayConfig,
+                                    Map<String, String> options,
+                                    String implementation) throws ServiceLifecycleException {
+        RemoteConfigurationMonitor service = null;
+        if (matchesImplementation(implementation, ZkRemoteConfigurationMonitor.class)) {
+            service = new ZkRemoteConfigurationMonitor(gatewayConfig, gatewayServices.getService(ServiceType.REMOTE_REGISTRY_CLIENT_SERVICE));
+        } else if (matchesImplementation(implementation, DbRemoteConfigurationMonitor.class)) {
+            service = createDbBasedMonitor(gatewayConfig, gatewayServices.getService(ServiceType.ALIAS_SERVICE));
```

Comment:
The abstract class has a protected method called `getAliasService` for the same; please re-use that.